cmd/clusterctl/examples/vsphere/generate-yaml.sh

```diff
@@ -106,6 +116,23 @@ kustomize build $VSPHERE_CLUSTER_API_CRD_PATH/default/ > $PROVIDERCOMPONENT_GENE
 echo "---" >> $PROVIDERCOMPONENT_GENERATED_FILE
 kustomize build $CLUSTER_API_CRD_PATH/default/ >> $PROVIDERCOMPONENT_GENERATED_FILE
 
+if [ $USE_KUBECON = 1 ]; then
+  echo "Updating generated provider-components.yaml file with containers from kubecon 2018..."
+#  CONTROLLER_CONTAINER=$(grep cluster-api-controller ${PROVIDERCOMPONENT_GENERATED_FILE})
+  CONTROLLER_CONTAINER=`grep "image:.*cluster-api-controller" ${PROVIDERCOMPONENT_GENERATED_FILE} | awk '{ print $2 }'`
+  VSPHERE_PROVIDER_CONTAINER=`grep "image:.*vsphere-cluster-api" ${PROVIDERCOMPONENT_GENERATED_FILE} | awk '{ print $2 }'`
+
+  cat $PROVIDERCOMPONENT_GENERATED_FILE | \
```

Comment:
line#125-133 can be re-written in a single command
`sed -i -e "s|$VSPHERE_PROVIDER_CONTAINER|$KUBECON_VSPHERE_PROVIDER|" -e "s|$CONTROLLER_CONTAINER|$KUBECON_CONTROLLER|" $PROVIDERCOMPONENT_GENERATED_FILE`

Reply:
I answered doug above around this topic.